modules/images/webp-uploads/load.php

```diff
@@ -573,3 +575,46 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
 	return webp_uploads_img_tag_update_mime_type( $html, 'post_thumbnail_html', $attachment_id );
 }
 add_filter( 'post_thumbnail_html', 'webp_uploads_update_featured_image', 10, 3 );
+
+/**
+ * Returns an array of image sizes that can have alternative mime types. By default, only core sizes
+ * are allowed and will be returned.
+ *
+ * If a custom size wants to have additional mime types support, they will need to have
+ * the provide_additional_mime_types property in the size details or it should be added using the
+ * webp_uploads_image_sizes_with_additional_mime_type_support filter.
+ *
+ * @since n.e.x.t
+ *
+ * @return array An array of image sizes that can have additional mime types.
+ */
+function webp_uploads_get_image_sizes_with_additional_mime_type_support() {
+	$additional_sizes = wp_get_additional_image_sizes();
+	$allowed_sizes    = array(
+		'thumbnail'      => 'thumbnail',
+		'medium'         => 'medium',
+		'medium_large'   => 'medium_large',
+		'large'          => 'large',
+		'post-thumbnail' => 'post-thumbnail',
+	);
+
+	foreach ( $additional_sizes as $size => $size_details ) {
+		if (
+			isset( $size_details['provide_additional_mime_types'] ) &&
+			filter_var( $size_details['provide_additional_mime_types'], FILTER_VALIDATE_BOOLEAN )
+		) {
```

Comment:
```suggestion
		if ( ! empty( $size_details['provide_additional_mime_types'] ) ) {
```

WP Core doesn't use `filter_var()` because it's not guaranteed to be available and/or accurate. If the intent of the performance lab plugin is to be merged in to core then it's best to avoid to reduce refactoring for the WP merge.

If this code isn't eventually intended for core then what you have is probably fien.

Reply:
@eugene-manuilov if there is a need to check that the value is boolean, could `is_bool` be used here instead?

Reply:
The idea was to allow developers to use `on`/`off`, `yes`/`no`, `1`/`0`, and `true`/`false` as possible values for the `provide_additional_mime_types` argument. Updated to check whether that argument is not empty.